lib/mysql.js

```diff
@@ -200,17 +200,8 @@ MySQL.prototype.executeSQL = function(sql, params, options, callback) {
   }
 };
 
-/**
- * Update if the model instance exists with the same id or create a new instance
- *
- * @param {String} model The model name
- * @param {Object} data The model instance data
- * @param {Function} [callback] The callback function
- */
-MySQL.prototype.updateOrCreate = MySQL.prototype.save =
-  function(model, data, options, callback) {
-    var fields = this.buildFields(model, data);
-
+ MySQL.prototype.modifyOrCreate = function(model, data, options, buildFields ,callback) {
+    var fields = buildFields(model, data);
```

Comment:
>Seem overly complicated passing in the functions.

Do think so? IMO it is not really complicated. I was following what you propose; please see [this comment](https://github.com/strongloop/loopback-connector-mysql/pull/139#discussion-diff-47161204)